File: app/src/main/java/igrek/todotree/service/history/LinkHistoryService.java
package igrek.todotree.service.history;

import java.util.HashMap;
import java.util.Map;

import igrek.todotree.domain.treeitem.AbstractTreeItem;
import igrek.todotree.domain.treeitem.LinkTreeItem;

public class LinkHistoryService {
	
	private Map<AbstractTreeItem, LinkTreeItem> target2link = new HashMap<>();
	
	public LinkHistoryService() {
	}
	
	public void storeTargetLink(AbstractTreeItem target, LinkTreeItem link) {
		target2link.put(target, link);
	}
	
	public LinkTreeItem getLinkFromTarget(AbstractTreeItem target) {
		return target2link.get(target);
	}
	
	public boolean hasLink(AbstractTreeItem target) {
		return target2link.containsKey(target);
	}
	
	public void resetTarget(AbstractTreeItem target) {
		target2link.remove(target);
	}
}

File: app/src/main/java/igrek/todotree/service/history/LinkHistoryService.kt
package igrek.todotree.service.history

import igrek.todotree.domain.treeitem.AbstractTreeItem
import igrek.todotree.domain.treeitem.LinkTreeItem
import java.util.HashMap

class LinkHistoryService {

    private val target2link: MutableMap<AbstractTreeItem, LinkTreeItem> = HashMap()

    fun storeTargetLink(target: AbstractTreeItem, link: LinkTreeItem) {
        target2link[target] = link
    }

    fun getLinkFromTarget(target: AbstractTreeItem): LinkTreeItem? {
        return target2link[target]
    }

    fun hasLink(target: AbstractTreeItem): Boolean {
        return target2link.containsKey(target)
    }

    fun resetTarget(target: AbstractTreeItem) {
        target2link.remove(target)
    }

    fun clear() {
        target2link.clear()
    }
}